client/data/site-settings/with-site-settings.js

```diff
@@ -0,0 +1,151 @@
+import { useQueryClient } from '@tanstack/react-query';
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { useCallback, useState } from '@wordpress/element';
+import { useTranslate } from 'i18n-calypso';
+import PropTypes from 'prop-types';
+import { useDispatch, useSelector } from 'react-redux';
+import useSiteSettingsQuery from 'calypso/data/site-settings/use-site-settings-query';
+import useUpdateSiteSettingsMutation from 'calypso/data/site-settings/use-update-site-settings-mutation';
+import { useProtectForm } from 'calypso/lib/protect-form';
+import { recordGoogleEvent, recordTracksEvent } from 'calypso/state/analytics/actions';
+import { errorNotice, successNotice } from 'calypso/state/notices/actions';
+import getCurrentRouteParameterized from 'calypso/state/selectors/get-current-route-parameterized';
+
+const noticeOptions = {
+	duration: 10000,
+	id: 'site-settings-save',
+};
+
+const withSiteSettings = createHigherOrderComponent( ( Wrapped ) => {
+	const WithSiteSettings = ( props ) => {
+		const { siteId } = props;
+		const [ unsavedSettings, setUnsavedSettings ] = useState( {} );
+		const dispatch = useDispatch();
+		const translate = useTranslate();
+		const queryClient = useQueryClient();
+		const path = useSelector( ( state ) => getCurrentRouteParameterized( state, siteId ) );
+		const { markChanged, markSaved } = useProtectForm();
+
+		const trackEvent = useCallback(
+			( name ) => {
+				dispatch( recordGoogleEvent( 'Site Settings', name ) );
+			},
+			[ dispatch ]
+		);
+
+		const trackTracksEvent = useCallback(
```

Comment:
Not sure these wrappers add much value. I'd likely just use `dispatch( recordTracksEvent(...) )`.